system/lib/libc/musl/include/stdbool.h

```diff
@@ -3,8 +3,11 @@
 
 #ifndef __cplusplus
 
-#define true 1
-#define false 0
+// These work for both C and javascript.
+// In C !!0 ==> 0 and in javascript !!0 ==> false
+// In C !!1 ==> 1 and in javascript !!1 ==> true
+#define true (!!1)
+#define false (!!0)
```

Comment:
When does this come into play though?  Do these macros expand inside of EM_JS/EM_ASM macros?  

I'd like to see a test that fails without this change if possible.

Reply:
They don't expand inside of them by default, but they do if you define another macro wrapping `EM_JS`. I can add a test.